### DISCUSSION TITLE
Supress grain reconstruction over resin

### DISCUSSION THREAD

**Discussion by Marie3133** [mtex-toolbox/mtex#792] [Ask Anything]
Hello everyone,

I am working on additive manufactured thin walls of Ti-6Al-4V, with an Widmenstatten microstructure (heat treatment at 920°C for 2h). My aim is to reconstruct prior beta grains of several samples.

I am using the code doc>Grains>GrainsReconstruction. The code runs successfully.
However, as my thin walls are of 200µm to 500µm, some areas of the scanned field aren't of titanium but of mounting resin. Thus, I get a wrong output on these areas (voronoi), see attached image.

Can somebody help me to understand how can I change the initial code to prevent the reconstruction of grains in  mounting resine areas please?

Best regards,
Marie

![1](https://user-images.githubusercontent.com/79309625/108490832-79955e80-72a3-11eb-8dbc-915cda6e9abb.png)

**Comment by kilir** [comment]
Hi,
please have a look at the documentation on grain reconstruction.
https://mtex-toolbox.github.io/GrainReconstruction.html#11

Cheers,
Rüdiger

**Reply by Marie3133** [reply]
My apologies, i am not using the GrainReconstruction code but the **alphaBetaTitanium** code for parent Beta Phase Reconstruction in Titanium Alloys (https://mtex-toolbox.github.io/TiBetaReconstructionAdvanced.html)

The GrainReconstruction code that you indicated uses  "grains = calcGrains(ebsdP('indexed'),'boundary','tight')"  to avoid grain reconstruction on non-indexed zones.

Unfortunately, I can not find any documentation about this for alphaBetaTitanium.

**Reply by ralfHielscher** [reply]
It is completely ok to use 
```
grains = calcGrains(ebsdP('indexed'),'boundary','tight')
```
with the alpha beta reconstruction code. All commands of MTEX can be combined with each other :)

Ralf.

**Comment by Marie3133** [comment]
Hi,

I am getting back to you because I still don't understand how can I deal with my problem here.

I tried two solutions using ; 
- The MTEX-5.6.0 code downloaded with the package 5.6.0 (with 'jobs');
- The TiBetaReconstruction code detailed on the online documentation.

1. I modified the  MTEX-5.6.0 code as you adviced :
```
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',1.5*degree,...
  'removeQuadruplePoints','boundary','tight')
```

![capture_jobs_boundary_tight](https://user-images.githubusercontent.com/79309625/108961035-1165c480-7677-11eb-87af-1b76f90b363f.png)

2. Same for the online documentation code  :
```
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'threshold',1.5*degree,...
  'removeQuadruplePoints','boundary','tight')
```

![tuto_boundary_tight](https://user-images.githubusercontent.com/79309625/108963308-39a2f280-767a-11eb-9779-d44496523dc8.png)


I think my issue is that I have some indexed points in the non-indexed area of the resin. I tried to removed them with the 'unitcell' function.

![capture_unitcell](https://user-images.githubusercontent.com/79309625/108961349-80dbb400-7677-11eb-8e8e-fe07330952f6.png)

But it didn't work  :

```
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'unitCell','threshold',1.5*degree,...
  'removeQuadruplePoints','boundary','tight');
```

![capture_jobs_Unitcell_boundary_tight](https://user-images.githubusercontent.com/79309625/108961653-f21b6700-7677-11eb-99f2-c9ce2a80b8d8.png)


Did I use correctly these function ? 
Could you give me some further help please ?

**Reply by ralfHielscher** [reply]
Could you somehow send the data? It seems the option 'boundary','tight' is completely ignored ... strange

Ralf.

**Reply by Marie3133** [reply]
Thanks a lot.
It is possible indeed, but I would rather sent it to you directly than to post it here as it is part of my PhD research. Is there an email adress that I can send it to ?

Marie.

**Reply by ralfHielscher** [reply]
Sure, I completely understand this. My email address is very easy to find ;)

Ralf.

**Comment by ralfHielscher** [comment]
You can try the following to reduce the strange looking outer grains:
```matlab
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'boundary','tight',1)
% remove very small grains
ebsd(grains(grains.grainSize < 5)) = [];
% redo grain reconstruction
[grains,ebsd.grainId] = calcGrains(ebsd('indexed'),'boundary','tight',1)
```
The result will still far from perfect. You may want to remove all boundary grains as described at
https://mtex-toolbox.github.io/SelectingGrains.html#20

Having the MTEX grain reconstruction algorithm to perfectly deal with such complicated 
outer boundaries is a nice challenge for the future.

All the best,
Ralf.
